app/src/main/java/samyups/example/trainingss/ui/MainViewModelFactory.kt

```diff
@@ -0,0 +1,16 @@
+package samyups.example.trainingss.ui
+
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.ViewModelProvider
+import samyups.example.trainingss.data.MainRepository
+
+class MainViewModelFactory(private val mainRepository: MainRepository)
+    : ViewModelProvider.NewInstanceFactory() {
+
+    @Suppress("UNCHECKED_CAST")
+    override fun <T: ViewModel?> create(modelClass: Class<T>): T {
+        return MainViewModel(mainRepository) as T
+    }
+}
```

Comment:
Do you understand what's happening here, or did you copy/paste? lol.  I was surprised to see this here so I looked into it a little bit and it looks like it's necessary for the `ViewModel` pattern to work correctly. This type of pattern is called a `Factory` pattern, in case you didn't have a change to read more about it.